### PR TITLE
PYIC 3329 Switch newly created Cimit stub to Java 17

### PR DESCRIPTION
**Pull request by rpayal**
## Proposed changes
Switch newly created Cimit stub (all lambdas) to Java 17

### What changed
- Changes made to switch cimit-stub module to Java17
- Changes made to stub modules to bump gradle-8.2.1.

### Issue tracking

- [PYI-3329](https://govukverify.atlassian.net/browse/PYI-3329)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: di-ipv-cimit-stub/lambdas/get-contra-indicator-credential/build.gradle [hunk 1]
```diff
@@ -19,15 +21,21 @@ dependencies {
...
-	compileOnly "org.projectlombok:lombok:1.18.22"
-	annotationProcessor "org.projectlombok:lombok:1.18.22"
+	compileOnly "org.projectlombok:lombok:1.18.28"
+	annotationProcessor "org.projectlombok:lombok:1.18.28"
 
-	aspect "software.amazon.lambda:powertools-logging:1.15.0",
-			"software.amazon.lambda:powertools-tracing:1.15.0"
+	aspect "software.amazon.lambda:powertools-logging:1.16.1",
+			"software.amazon.lambda:powertools-tracing:1.16.1"
+	aspect('org.aspectj:aspectjrt') {
```
Comment (therealslimsladey) [inline di-ipv-cimit-stub/lambdas/get-contra-indicator-credential/build.gradle:29]: What's the reason for introducing this?

File: di-ipv-cimit-stub/lambdas/get-contra-indicator-credential/build.gradle [hunk 2]
```diff
@@ -19,15 +21,21 @@ dependencies {
...
+		version {
+			strictly '1.9.8'
+		}
+	}
 
 	testImplementation "org.junit.jupiter:junit-jupiter:5.9.3",
 			"org.mockito:mockito-junit-jupiter:5.4.0",
 			'uk.org.webcompere:system-stubs-jupiter:2.0.2'
+	testRuntimeOnly("org.junit.platform:junit-platform-launcher")
```
Comment (therealslimsladey) [inline di-ipv-cimit-stub/lambdas/get-contra-indicator-credential/build.gradle:38]: What's the reason for introducing this?

File: di-ipv-credential-issuer-stub/build.gradle [hunk 3]
```diff
@@ -10,7 +10,9 @@ repositories {
 	mavenCentral()
 }
 
-sourceCompatibility = 17
+java {
+	sourceCompatibility = JavaVersion.VERSION_17
```
Comment (therealslimsladey) [inline di-ipv-credential-issuer-stub/build.gradle:14]: Specify targetCompatibility as well to avoid doubt.

File: di-ipv-core-stub/build.gradle [hunk 4]
```diff
@@ -10,7 +10,9 @@ repositories {
 	mavenCentral()
 }
 
-sourceCompatibility = 17
+java {
+	sourceCompatibility = JavaVersion.VERSION_17
```
Comment (therealslimsladey) [inline di-ipv-core-stub/build.gradle:14]: And targetCompatibility

File: di-ipv-orchestrator-stub/build.gradle [hunk 5]
```diff
@@ -10,7 +10,9 @@ repositories {
 	mavenCentral()
 }
 
-sourceCompatibility = 17
+java {
+	sourceCompatibility = JavaVersion.VERSION_17
```
Comment (therealslimsladey) [inline di-ipv-orchestrator-stub/build.gradle:14]: and targetCompatibility

**Review by therealslimsladey — CHANGES_REQUESTED** [review]

File: di-ipv-credential-issuer-stub/build.gradle (hunk 3 above)
Comment (rpayal) [reply di-ipv-credential-issuer-stub/build.gradle:14]: By default, if not specified, targetCompatibility=sourceCompatibility.

File: di-ipv-cimit-stub/lambdas/get-contra-indicator-credential/build.gradle (hunk 1 above)
Comment (rpayal) [reply di-ipv-cimit-stub/lambdas/get-contra-indicator-credential/build.gradle:29]: To override the transient dependency, explained in the ticket comment.

Resolve transitive dependency issue for aspectjrtfrom software.amazon.lambda:powertools-logging and software.amazon.lambda:powertools-tracing to resolve following issue during build
> Task :di-ipv-cimit-stub:lambdas:get-contra-indicators:compileJava FAILED
[error] target level should be in '1.1'...'1.8','9'...'16' (or '5.0'..'16.0') or cldc1.1: 17
org.aspectj:aspectjrt to version 1.9.8

File: di-ipv-cimit-stub/lambdas/get-contra-indicator-credential/build.gradle (hunk 2 above)
Comment (rpayal) [reply di-ipv-cimit-stub/lambdas/get-contra-indicator-credential/build.gradle:38]: Added testRuntimeOnly("org.junit.platform:junit-platform-launcher") in build.gradle to remove the following deprecation warning;

Task :di-ipv-cimit-stub:lambdas:get-contra-indicator-credential:test
The automatic loading of test framework implementation dependencies has been deprecated. This is scheduled to be removed in Gradle 9.0. Declare the desired test framework directly on the test suite or explicitly declare the test framework implementation dependencies on the test's runtime classpath. Consult the upgrading guide for further information: [Upgrading your build from Gradle 8.x to the latest](https://docs.gradle.org/8.2.1/userguide/upgrading_version_8.html#test_framework_implementation_dependencies)

**Review by TalatAydinGDS — CHANGES_REQUESTED** [review]
These changes looks good for me. but I think, don't need to repeat the common configurations in each lambda projects 'build.gradle' file. We can add hierarchy in the root 'build.gradle' file to avoid duplicating the similar configuration for all lambdas projects. By doing this, we can make changes in one places and have those changes reflected in all lambda projects. So, this hierarchy will help we centralize the configurations and make it easier to manage and maintain our Gradle project and can see only different dependencies and configurations in lambda's build.gradle. We can do it in this task or different task?

**Review by TobySaundersGDS — APPROVED** [review]

**Review by therealslimsladey — APPROVED** [review]

**Review by TalatAydinGDS — APPROVED** [review]
LGTM

**Pull request merged**